daphne/src/messages.rs

```diff
@@ -389,36 +461,91 @@ impl Encode for Interval {
 impl Decode for Interval {
     fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
         Ok(Self {
-            start: u64::decode(bytes)?,
-            duration: u64::decode(bytes)?,
+            start: Time::decode(bytes)?,
+            duration: Duration::decode(bytes)?,
         })
     }
 }
 
+/// A query issued by the Collector in a collect request.
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Query {
+    TimeInterval { batch_interval: Interval },
+    FixedSize { batch_id: Id },
+}
+
+impl Query {
+    pub(crate) fn unwrap_interval(&self) -> &Interval {
+        if let BatchSelector::TimeInterval { ref batch_interval } = self {
+            batch_interval
+        } else {
+            panic!("TOOD(issue #100)");
```

Comment:
```suggestion
            panic!("TODO(issue #100)");
```
Panic with attitude!